src/t_stream.c

```diff
@@ -2247,6 +2247,17 @@ void xreadCommand(client *c) {
         return;
     }
 
+    for (int i = streams_arg + streams_count; i < c->argc; i++) {
+        for (int j = i + 1; j < c->argc; j++) {
```

Comment:
Small question which might not be super important- does that implies that the command is no longer O(NM) where N=number of keys and M is the number of msgs returned by each key, but rather O(N^2+NM)?

Reply:
good point, i'll update the json files

Reply:
considering N is expected to be rather low, maybe it shouldn't be there?
@itamarhaber?

Reply:
or if it isn't expected to be always rather low, maybe this change is wrong..

Reply:
also with the new blocking mechanism we will repeat this when reprocessing... might be able to optimize in case this is run during unblock?

Reply:
yes, we can skip that validation in that case, but in order to do that we'll probably need to add a new flag to the client struct, saying we are reprocessing after unblocking... sounds good?

i'm wondering if there are any other expensive validations/calculations we can skip in other commands... can you please have a quick look?

Reply:
I am sure we could spare some checks, but I will argue that every command should have arity check function (could even be generated automatically from json), so we could better way to separate the 2 calls and fail the command on invalid arguments at much earlier stage. However that is a big change, and I guess making this case O(2(N^2)+NM) is not the critical thing

Reply:
Shall we restrict this PR to just adding validation for duplicate key access in XREAD[GROUP] and probably file a separate issue/PR for avoiding revalidation during reprocessing ?

Reply:
yeah, we can.. that way we can maybe handle the re-validation optimization earlier.